File: Data Structures & Algorithms/max-water-container/submission-4.cs
public class Solution {
    public int MaxArea(int[] heights) {
        int maxVolume = 0;
        int left = 0;
        int right = heights.Length - 1;

        while(left < right){
            int width = right - left;
            int currentHeight = Math.Min(heights[left], heights[right]);
            int currentVolume = width * currentHeight;

            maxVolume = Math.Max(maxVolume, currentVolume);

            if(heights[left] < heights[right]){
                left++;
            } else {
                right--;
            }
        }
        return maxVolume;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-4.js
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    lengthOfLongestSubstring(s) {
        const set = new Set();
        let longestStr = 0;
        let left = 0;

        for(let right = 0; right < s.length; right++){

            while(set.has(s[right])){
                set.delete(s[left]);
                left++;
            }

            set.add(s[right]);

            longestStr = Math.max(longestStr, right - left + 1)
        }

        return longestStr
    }
}
